src/pages/test-report/components/competency/competency.ts

```diff
@@ -42,14 +47,27 @@ export class CompetencyComponent {
   competencyState: ComptencyState;
   subscription: Subscription;
   faultCount: number;
-
+  promoted: boolean;
```

Comment:
I don't really follow what `promoted` means in this context, perhaps we can come up with a better name?